db/collection.go

```diff
@@ -13,46 +13,69 @@ import (
 
 // Collection represents a set of a specific type of model.
 type Collection struct {
-	db        *DB
+	*readOnlyCollection
```

Comment:
@fabioberger in one of your PRs, I suggested that we don't need to use a pointer for embedded fields. Usually, this is true. However, in this specific case, we need to mutate `readOnlyCollection` and `writeOnlyCollection` when calling embedded methods and this is the only way to make it work.